py/examples/table_pagination_h2o3.py

```diff
@@ -0,0 +1,225 @@
+            q.client.filters = None
+            q.client.page_offset = 0
+            q.client.total_rows = len(q.app.h2o_df)
+
+        # Update the rows on in our UI
+        # TODO: if you change where your table is located, this needs updating
+        q.page["table_card"].items[0].table.rows = get_table_rows(q)
+        q.page["table_card"].items[0].table.pagination.total_rows = q.client.total_rows
+
+    await q.page.save()
+
+
+def get_table_rows(q: Q):
+    logger.info(
+        f"Creating new table for rows: {q.client.page_offset} to {q.client.page_offset + q.app.rows_per_page}"
+    )
+
+    working_frame = prepare_h2o_data(q)
+
+    # Bring our limited UI rows locally to pandas to prepare for our ui.table
+    local_df = working_frame[
+        q.client.page_offset : q.client.page_offset + q.app.rows_per_page, :
+    ].as_data_frame()
+    q.client.total_rows = len(working_frame)
+
+    table_rows = [
+        ui.table_row(
+            name=str(
+                q.client.page_offset + i
+            ),  # name is the index on the h2o dataframe for appropriate lookup
+            cells=[str(local_df[col].values[i]) for col in local_df.columns.to_list()],
+        )
+        for i in range(len(local_df))
+    ]
+
+    h2o.remove(working_frame)  # remove our duplicate work
+
+    return table_rows
+
+
+async def download_h2o_table(q: Q):
+    # Create a unique file name as this is a multi-user app
+    local_file_path = f"h2o3_data_{str(int(time()))}.csv"
+    working_frame = prepare_h2o_data(q)
+
+    h2o.download_csv(working_frame, local_file_path)
+    (wave_file_path,) = await q.site.upload([local_file_path])
+    os.remove(local_file_path)
+
+    q.page["meta"].script = ui.inline_script(f'window.open("{wave_file_path}")')
+
+
+def prepare_h2o_data(q: Q):
+
+    # This is used to prep the data we want to show on the screen or download, so it gets its own function
+    # If you have 5 users at the same time, there will be 6 large dataframes in h2o3 - ensure proper cluster size
+    working_frame = h2o.deep_copy(q.app.h2o_df, "working_df")
+
+    if q.client.sort is not None:
+        # H2O-3 can only sort numeric values - if the developer allows users to sort
+        # string columns the end users will see unexpected results
+
+        working_frame = working_frame.sort(
+            by=list(q.client.sort.keys()), ascending=list(q.client.sort.values())
+        )
+
+    if q.client.filters is not None:
+
+        for key in q.client.filters.keys():
+            working_frame = working_frame[
+                working_frame[key].match(q.client.filters[key])
+            ]
+
+    if q.client.search is not None:
+        # We check if our search term is in any of the searchable columns
+        # Start with and index of 0s and then filter to only keep rows with index > 0
+
+        index = h2o.create_frame(
+            rows=len(working_frame), cols=1, integer_fraction=1, integer_range=1
+        )
+        index["C1"] = 0
```

Comment:
This basically creates an empty array of 0's that is the size of the H2O Frame and then for each "searchable" column we update the array with + 1 if the item is there. So that the search result isn't only valid it it's in **all** columns but just at least one. 

So at the end each index of the frame with have 0 or >0 and we return rows >0.

There may be a more effecient way to do this! but this is what I could come up with